runtime/js-compute-runtime/core/event_loop.cpp

```diff
@@ -301,7 +305,11 @@ bool EventLoop::process_pending_async_tasks(JSContext *cx) {
   if (builtins::Request::is_instance(ready_obj)) {
     ok = process_pending_request(cx, ready_obj, host_api::HttpPendingReq{ready_handle});
   } else if (builtins::KVStore::is_instance(ready_obj)) {
-    ok = builtins::KVStore::process_pending_kv_store_lookup(cx, ready_obj);
+    if (builtins::KVStore::has_pending_delete_handle(ready_obj)) {
+      ok = builtins::KVStore::process_pending_kv_store_delete(cx, ready_obj);
+    } else {
+      ok = builtins::KVStore::process_pending_kv_store_lookup(cx, ready_obj);
+    }
```

Comment:
Note: Something needs to be thought about here -- the KVStore instance could have both a pending lookup and a pending delete, currently this implementation does not correctly allow for that

Reply:
Could you add a comment on this, or make an issue to track that we need to handle this better?

Reply:
> Could you add a comment on this, or make an issue to track that we need to handle this better?

I've got a second pull request open (in draft right now) which will solve this and fix two other bugs in our async kvstore operations 🙈